libraries/habitat_shared.rb

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright:: 2017-2018 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# li
+
+module Habitat
+  module Shared
+    HAB_VERSION = '0.67.0'.freeze
+    LINUX_LAUNCHER_VERSION = '8659'.freeze
+    WINDOWS_LAUNCHER_VERSION = '8656'.freeze
+    WINDOWS_SERVICE_VERSION = '0.3.1'.freeze
+
+    def hab_version
+      HAB_VERSION
+    end
+
+    def hab_launcher_version
+      if platform_family?('windows')
+        WINDOWS_LAUNCHER_VERSION
+      else
+        LINUX_LAUNCHER_VERSION
+      end
+    end
+
+    def hab_windows_service_version
+      WINDOWS_SERVICE_VERSION
+    end
+
+    def hab(*command)
+      # Windows shell_out does not support arrays, so manually cleaning and joining
+      hab_cmd = (['hab'] + command).flatten.compact.join(' ')
```

Comment:
actually this should probably only be done on windows.  there's benefits on the unix-side to passing through the array version since then strings-with-spaces don't need to be quoted.

still since we can't do this on windows that means that every argument which could be a file-with-spaces ("C:\Program Files" trivially triggers these bugs) needs to be quoted.

Reply:
(i'm less concerned with fixing every last issue in the latter case since going from no-windows support to some-bugs-in-windows support is still progress, but the unix API shouldn't go backwards in the process)

Reply:
@lamont-granquist I made the assignment of hab_cmd conditional on platform